fix(#3804): usePopover: non-modal popover allows page and popover to scroll when target has position: fixed

majornista opened the pull request:
Closes #3804

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue #3804](https://github.com/adobe/react-spectrum/issue/3804).
- [x] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

<!--- Include instructions to test this pull request -->

## 🧢 Your Project:

Adobe/Accessibility

LFDanLu reviewed (commented):
Tested in VO and Talkback, confirmed that I was still able to interact with the ComboBox even with the dropdown open with these changes. However, I was no longer able to scroll the list of options on iPad [here](https://reactspectrum.blob.core.windows.net/reactspectrum/0ee71a531fcce40bbb58a4e0954f8902a23b117e/docs/react-spectrum/ComboBox.html), most likely due to how usePreventScroll prevents default on certain touch events. Additionally, due to the scroll position handling that happens, the dropdown extends past the edge of the screen on iPad (though we have a different issue without this change due to the virtual keyboard that appears)

I'll need to think about what we can do here instead or if we need to allow position: fixed to be provided to a overlay

majornista commented:
@LFDanLu I agree that we will probably need to account for targets that have or inherit `position: fixed` by making the overlay also have `position: fixed` and by taking that into account when positioning the overlay.

LFDanLu commented:
Hm, definitely weird that the storybook example works on iPad but it doesn't in the docs (maybe something to do with the storybook not being scrollable?). Maybe indicative of a bug in the usePreventScroll code since it should allow scrolling in nested scroll regions now that I'm looking at the code again

ktabors commented on packages/@react-aria/overlays/src/usePopover.ts:
```diff
@@ -92,8 +92,18 @@ export function usePopover(props: AriaPopoverProps, state: OverlayTriggerState):
     onClose: null
   });
 
+  // Delay preventing scroll until popover is positioned to avoid extra scroll padding.
+  // This requires a layout effect so that positioning has been committed to the DOM
+  // by the time usePreventScroll measures the element.
+  let [isPositioned, setPositioned] = useState(false);
+  useLayoutEffect(() => {
+    if (placement) {
+      setPositioned(true);
+    }
+  }, [isNonModal, placement]);
+
```
```suggestion
```
Don't think we need the useLayoutEffect.

ktabors commented on packages/@react-aria/overlays/src/usePopover.ts:
```diff
@@ -92,8 +92,18 @@ export function usePopover(props: AriaPopoverProps, state: OverlayTriggerState):
+
   usePreventScroll({
-    isDisabled: isNonModal
+    isDisabled: !isPositioned
```
```suggestion
```
We'd like to test this PR with the useLayoutEffect removed and usePreventScroll() always set. Looking at related PRs we think the issues are related to iPad async example.
